code-challenges.test.js

```diff
@@ -16,47 +16,272 @@
+//had a lot of fun with this one.
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-var arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-var letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-var letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+//JEST TEST: testing function containsChar that takes in an array of words and a single letter and returns all the words that contain that letter in an array
+describe("testing function containsChar", () => {
+    it("should take in array and specific letter and return the words that contain that letter in an array", () => {
+
+        var arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+        var letterA = "a"
+        // Expected output: ["Apple", "Banana", "Orange"]
+        var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+        var letterE = "e"
+        // Expected output: ["Cherry", "Blueberry", "Peach"]
+        
+        expect(containsChar(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+        expect(containsChar(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+  })
+  //test fails because: ReferenceError: containsChar is not defined
 
 
 
 // b) Create the function that makes the test pass.
 
+//create containsChar that takes in array of words and a letter
+// const containsChar = (array, char) =>{
+//     //create new array to push to
+//     let doesContain = [];
+
+//     //iterate through array 
+//     for (let i = 0; i < array.length; i++) {
+//         //test whether each word contains the letter (lower or capital) in the string
+//         if(array[i].includes(char.toUpperCase()) || array[i].includes(char.toLowerCase())){
+//             //if contains either, push word to array
+//             doesContain.push(array[i]);
+//         }
+//         // console.log(doesContain); 
+//     }
+//     //return new array
+//     return doesContain;
+// }
+//test had a couple issues syntax wise but after fixing some parenthesis it passes!
+
+//REFACTOR WITH .filter()! ===============================================================
+
+//create containsChar that takes in array of words and a letter
+const containsChar = (array, char) =>{
+    //returns a filtered array
+        //filter looks at word and if it includes char either upper or lower case then return true and add to new array
+    return array.filter(word => 
+            word.includes(char.toUpperCase()) || 
+            word.includes(char.toLowerCase())
+        )
+}
+//test passed!
```

Comment:
Great work here! Understanding whats happening with filter by writing it out long hand is useful practice!